src/Compiler/Checking/NicePrint.fs

```diff
@@ -2225,6 +2227,18 @@ module TastDefinitionPrinting =
             else 
                 (lhsL ^^ WordL.equals) -* rhsL
 
+        let tryGetFieldXml (layoutList: Layout list) idxOpt =
+            match idxOpt with
+            | Some i ->
+                let t = layoutList[i]
```

Comment:
Perharps add an defensive check here, to prevent possible out of bounds in case of degenerate input?